File: src/entitie/hazem/SalleDeCinema.java
package entitie.hazem;

public class SalleDeCinema {

    private int id;
    private String nom;
    private int nbrPlace;
    private String Description;
    private String adresse;
    private int idUser;

    public SalleDeCinema() {
    }

    public SalleDeCinema(int id, String nom, int nbrPlace, String description, String adresse, int idUser) {
        this.id = id;
        this.nom = nom;
        this.nbrPlace = nbrPlace;
        Description = description;
        this.adresse = adresse;
        this.idUser = idUser;
    }

    public SalleDeCinema(String nom, int nbrPlace, String description, String adresse, int idUser) {
        this.nom = nom;
        this.nbrPlace = nbrPlace;
        Description = description;
        this.adresse = adresse;
        this.idUser = idUser;
    }

    public SalleDeCinema(int id, String nom, int nbrPlace, String description, String adresse) {
        this.id = id;
        this.nom = nom;
        this.nbrPlace = nbrPlace;
        Description = description;
        this.adresse = adresse;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public int getNbrPlace() {
        return nbrPlace;
    }

    public void setNbrPlace(int nbrPlace) {
        this.nbrPlace = nbrPlace;
    }

    public String getDescription() {
        return Description;
    }

    public void setDescription(String description) {
        Description = description;
    }

    public String getAdresse() {
        return adresse;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }

    public int getIdUser() {
        return idUser;
    }

    public void setIdUser(int idUser) {
        this.idUser = idUser;
    }

    @Override
    public String toString() {
        return "SalleDeCinema{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", nbrPlace=" + nbrPlace +
                ", Description='" + Description + '\'' +
                ", adresse='" + adresse + '\'' +
                ", idUser=" + idUser +
                '}';
    }
}

File: src/Entity/SalleDeCinema.php
<?php

namespace App\Entity;

use App\Repository\SalleDeCinemaRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=SalleDeCinemaRepository::class)
 */
class SalleDeCinema
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\Length(
     * min = 3,
     * max = 50,
     * minMessage = "Le nom de salle doit comporter au moins {{ limit }} caractères",
     * maxMessage = "Le nom de salle doit comporter au plus {{ limit }} caractères"
     * )
     */
    private $nom;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank
     * @Assert\Range(
     *      min = 10,
     *      max = 250,
     *      notInRangeMessage = "You must be between {{ min }} and {{ max }}  to enter",
     * )
     */
    private $nbrPlace;

    /**
     * @ORM\Column(type="text")
     * @Assert\NotBlank
     *
     */
    private $Description;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     */
    private $adresse;

    /**
     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="salleDeCinemas")
     * @ORM\JoinColumn(nullable=false)
     */
    private $user;

    /**
     * @ORM\OneToMany(targetEntity=Reclamation::class, mappedBy="salle")
     */
    private $reclamations;


    public function __construct()
    {
        $this->reclamations = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getNbrPlace(): ?int
    {
        return $this->nbrPlace;
    }

    public function setNbrPlace(int $nbrPlace): self
    {
        $this->nbrPlace = $nbrPlace;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->Description;
    }

    public function setDescription(string $Description): self
    {
        $this->Description = $Description;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    /**
     * @return Collection|Reclamation[]
     */
    public function getReclamations(): Collection
    {
        return $this->reclamations;
    }

    public function addReclamation(Reclamation $reclamation): self
    {
        if (!$this->reclamations->contains($reclamation)) {
            $this->reclamations[] = $reclamation;
            $reclamation->setSalle($this);
        }

        return $this;
    }

    public function removeReclamation(Reclamation $reclamation): self
    {
        if ($this->reclamations->removeElement($reclamation)) {
            // set the owning side to null (unless already changed)
            if ($reclamation->getSalle() === $this) {
                $reclamation->setSalle(null);
            }
        }

        return $this;
    }

}
